### PR TITLE
Create ViewStateModel to support help/exit commands [Model]

### LINKED ISSUE #155: Create UIStateModel
Currently, our model can be separated into two:
1. One is business-logic related, such as `BlockModel` and `SuggestionModel`
2. Another is UI-state related, such as `CommandInputModel`

Let's aggregate both the `StringProperty commandInputProperty` from `CommandInputModel` and this help flag property into a new model class, called `UIStateModel`. that means, this `UIStateModel` is going to expose a `BooleanProperty` indicating whether or not the help modal should be opened and a `StringProperty` indicating the current user input

### PR DESCRIPTION
**Pull request by HemanshuGandhi**
Solves #155 

Summary of Changes:
1. Creates a new package "viewstate" within the "model" package, which now contains all UI-related models.
2. Creates viewstate/`HelpFlagModel` interface. 
3. Creates viewstate/`ViewStateModel` interface that aggregates `HelpFlagModel` and `CommandInputModel` interfaces.
4. Makes ` Model` extends `ViewStateModel`.
5. Removes the `CommandInputModelImpl` class and creates a `ViewStateModelImpl` that implements methods from both interfaces. 
6. Exposes `ViewStateModel` to `ModelManager`(instead of `CommandInputModel`)
7. Updates `ModelStubBase` accordingly.

### CODE REVIEW

**Comment by HemanshuGandhi** [comment]
@ljiazh3ng you can update `BoolProperty` within Model's `ViewStateModel` (which comes from the `HelpFlagModel`) whenever "Help" command is invoked. I will set the UI to listen to changes to this BoolProperty, and pop up the Help Modal accordingly.

**Comment by kevinputera** [comment]
@HemanshuGandhi It seems that there was some merging error previously! My commits for `AbsolutePathCorrectionEngine` appears in this PR. I rebased this PR branch to the latest master. Hope you don't mind!

File: src/main/java/com/notably/model/viewstate/HelpFlagModel.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+     * Gets the flag representing whether the Help Modal should be displayed or not.
+     * @return A Boolean. true if the Help Model should be displayed, false otherwise.
+     */
+    Boolean getBool();
+
+    /**
+     * Sets the value for the boolean flag representing whether the Help Modal should be displayed or not.
+     * @param bool A String corresponding to the new text data in the CommandInputModel.
+     */
+    void setBool(Boolean bool);
```
Comment (kevinputera) [inline src/main/java/com/notably/model/viewstate/HelpFlagModel.java]: Hmmm calling it `getBool` doesn't sound very descriptive 😅. How about we call it `isHelpOpen`? We can rename `boolProperty` -> `helpOpenProperty` and `setBool` -> `setHelpOpen` too!

**Review by kevinputera — APPROVED** [review]
Looks good to me! I'm gonna approve this so that you can merge it in once you have made your final changes. Only some comments regarding naming below!

File: src/main/java/com/notably/model/viewstate/HelpFlagModel.java (hunk 1 above)
Comment (HemanshuGandhi) [reply src/main/java/com/notably/model/viewstate/HelpFlagModel.java]: Alright, will do so.

**Pull request merged**